src/event-bus/EventStore.ts

```diff
@@ -0,0 +1,106 @@
+import { Event, PersistedEvent } from '../shared/events';
+import { Logger } from '../shared/logger';
+import { promisify } from '../shared/promisify';
+import { sleep } from '../shared/sleep';
+
+const eventStoreConstructor = require('eventstore');
+
+export default class EventStore {
+
+  private isConnected = false;
+  private eventStore: any;
+
+  constructor(
+    private readonly publisher: (event: Event<any>, cb: (err: any) => void) => void,
+    private readonly eventStoreConfig: any,
+    private readonly logger: Logger,
+    private readonly numRetries = 1,
+    private readonly sleepFor = 300,
+  ) {
+  }
+
+  public async persistEvent(event: Event<any>) {
+    const tryPersistEvent = async () => {
```

Comment:
It's a good idea to [avoid function declarations inside functions](https://www.reaktor.com/blog/javascript-performance-fundamentals-make-bluebird-fast/). Unless I'm mistaken, you could make this function a member of the class and pass in the `event` to be persisted instead of using it from the closure.

This, of course, has a rather negligible effect on our performance, but it's probably still a good idea because the function signature explicitly tells us what it's doing.